src/app/components/elements/ScrollShadows.tsx

```diff
@@ -47,10 +47,12 @@ export const ScrollShadows = <T extends HTMLElement>({ element }: { element: T |
   useEffect(() => {
     if (!element) return;
     const resizeObserver = new ResizeObserver((entries) => {
-      const element = entries[0].target;
-      setClientWidth(element.clientWidth);
-      setScrollWidth(element.scrollWidth);
-      setScrollLeft(element.scrollLeft);
+      setTimeout(() => {
```

Comment:
Could you elaborate on this change, please? It looks a bit hacky, given that the timeout is 0.

Reply:
Yeah this one wasn't ideal. Essentially without this I get occasional "ResizeObserver loop limit exceeded" errors, when using the expand/close button. I did some googling and found it's a relatively common problem, here was one of the explanations: 

`The root of this problem seems to be React re-rendering (or any computationally intensive work, really) happening as a direct result of the resize event. If you defer that work to after the resize event has been handled, the browser is totally fine with it because it's not being prevented from delivering all of its event notifications in time.`

The most common workaround seems to be the setTimeout 0 option, so that React handles it at the end and avoids errors. Using requestAnimationFrame() was also a suggested fix but that one caused other side effects. Debounce/throttle were also suggested fixes. I've manually tested it quite a bit and this timeout seems to not have caused any unexpected behaviour, and it fixed the error.

I did see some suggestions that the error won't happen once we are using react-dom 18, but we are currently on 17.

Reply:
Thank you for the detailed explanation! That makes sense. I think this is one of those rare occasions in which I would definitely recommend adding a comment with your explanation, because otherwise a developer won't be able to figure out why the timeout is needed just by looking at the code. If we already have a ticket for upgrading React, it would be good to mention in that ticket that we should revisit this then and check if we can remove the workaround.